compiler/llvm/llvmAggregateGlobalOps.cpp

```diff
@@ -78,11 +77,25 @@
 #include <list>
 #include <string>
 #include <unordered_map>
+#include <memory>
+
+#if HAVE_LLVM_VER >= 160
+#include <optional>
+#else
+#include "llvm/ADT/Optional.h"
+#endif
 
 using namespace llvm;
 
 namespace {
 
+template <typename T>
+#if HAVE_LLVM_VER >= 160
+using optional = std::optional<T>;
+#else
+using optional = llvm::Optional<T>;
+#endif
+
```

Comment:
I just happened to stumble upon the PR and I know it's a draft, so apologies if this is premature.

Can this code just reference `chpl::optional` from `frontend`? It would be nice to reduce duplication there.